extensions/scarb-doc/src/metadata/compilation.rs

```diff
@@ -131,10 +135,30 @@ fn get_crate_settings_for_package(
             .contains(&String::from("coupons")),
     };
 
+    let dependencies = package
+        .dependencies
+        .iter()
+        .map(|dependency| {
+            let version = packages
+                .iter()
+                .find(|package| package.name == dependency.name)
+                .unwrap()
```

Comment:
I would rather do `compilation_unit_metadata .components` instead of `metadata.packages` here and skip the package if find returns `None`. My point is, that the `components` is what is fed to the compiler as crate roots, not `metadata.packages` (which can be a superset of `components`).

Reply:
Need to use packages because the compilation unit component has not version info.